src/listpack_malloc.h

```diff
@@ -39,8 +39,10 @@
 #ifndef LISTPACK_ALLOC_H
 #define LISTPACK_ALLOC_H
 #include "zmalloc.h"
-#define lp_malloc zmalloc
-#define lp_realloc zrealloc
+#define lp_malloc zmalloc_safe
+#define lp_realloc zrealloc_safe
 #define lp_free zfree
+/* todo: There is no need to use zmalloc_usable_size_safe() because we have
+ * already ensured that it is safe for listpack to be allcated or reallocated */
```

Comment:
that comment was a little bit unclear to me at first, so let me start from the beginning.

we currently have two patterns in the code:
1. calling a plain `zmalloc` or `zrealloc` and then calling `zmalloc_usable_size` and using that memory.
2. calling `zmalloc_usable` and using the size it returns.

most of the places can be converted to use approach [2] as you did, and we need to avoid using [1] if we can.
it would have been nice to completely obliterate zmalloc_usable_size, but since we can't we need to document that anyone who uses it needs to also call extend_to_usable manually.

let's document that in zmalloc.h, and also as you did here (no `todo:` though, and no zmalloc_usable_size_safe which i suggest to drop, see other comment)

Reply:
i see you reverted all changes in listpack. 
don't we need to use zmalloc_usable in listpack_malloc.h (and some comment explaining it)?
i.e. same as we did in module.c

Reply:
Yes, because I made `zmalloc_usable_size()` safe, and I think `lp_malloc_size()` will be called much less frequently than malloc/realloc in listpack.
Do you mean we should avoid approach  [2] ultimately? I'll modify it, I need to benchmark them.

Reply:
Done in https://github.com/redis/redis/pull/11982/commits/c3b4c6f763abe800680801eb55a68680d6519bf4